homeassistant/components/axis/config_flow.py

```diff
@@ -156,21 +156,21 @@ async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowRes
         """Prepare configuration for a DHCP discovered Axis device."""
         return await self._process_discovered_device(
             {
-                CONF_HOST: discovery_info[dhcp.IP_ADDRESS],
-                CONF_MAC: format_mac(discovery_info[dhcp.MAC_ADDRESS]),
-                CONF_NAME: discovery_info[dhcp.HOSTNAME],
+                CONF_HOST: discovery_info.ip,
+                CONF_MAC: format_mac(discovery_info.macaddress),
+                CONF_NAME: discovery_info.hostname,
                 CONF_PORT: DEFAULT_PORT,
             }
         )
 
-    async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo):
+    async def async_step_ssdp(self, discovery_info: ssdp.SsdpServiceInfo) -> FlowResult:
         """Prepare configuration for a SSDP discovered Axis device."""
-        url = urlsplit(discovery_info["presentationURL"])
+        url = urlsplit(discovery_info.upnp["presentationURL"])
```

Comment:
Why is it upnp and not ssdp?

Reply:
@chism please correct me if I am explain it wrongly, as I may have to explain it in multiple integrations.

The `ssdp` discovery works by combining the result of two broadcasts :
- one broadcast/response contains only SSDP data
- another broadcast/response contains only UPnP data

The full data is now available in two separate fields: `discovery_info.ssdp_headers` and `discovery_info.upnp`, with some of the SSDP data extracted into the various `discovery_info.ssdp_XXX` fields.

In the case of `axis`, I believe `presentationURL`, `serialNumber` and `friendlyName` are part of the UPnP data not part of the `SSDP` data but I could be wrong and it would be great if you had the opportunity to check it.

Reply:
There is probably a better source than Microsoft for this, but this gives an indication of the fields that are provided by UPnP: https://docs.microsoft.com/en-us/windows/win32/api/upnp/nn-upnp-iupnpdevice

Reply:
I have gone even further on the PR, and use constants for the UPNP information. This highlights the fact that it is indeed UPnP data and not part of the SSDP headers.
In my opinion this is ready to merge.